modules/codegen-plugin/src/smithy4s/codegen/Smithy4sCodegenPlugin.scala

```diff
@@ -290,6 +314,59 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
       )
     )
 
+  private def makeSmithyBuildMerge(
+      config: Configuration
+  ): Def.Initialize[Task[File]] = Def.task {
+    val sbFile = (config / smithy4sSmithyBuildFile).value
+
+    val smithyBuildJson = makeSmithyBuild(config).value
+
+    synchronized {
```

Comment:
it seems like `task / aggregate := false` did not prevent parallel execution as I had hoped and sometimes I was left with a broken json file. so I slapped a synchronized

Reply:
tbh that sounds like a bit of a band aid rather than a long-term solution. I'd rather have one top-level task that you run on your `root` project, which eliminates any concurrency in side-effecting IO

Reply:
we discussed this internally, and I don't know how to implement such a task

especially not in a sbt plugin

maybe @keynmol would know?

context: I'd like a task that I can run on the root module (or any aggregate module) that runs a task defined in a plugin. ideally, the task would aggregate the result of all the task executions, and then write to a file. the idea is that every module that has `Smithy4sCodegenPlugin` has a task to generate a `smithy-build.json` definition, but we'd like to have one task to reduce them all together and write them to disk

Reply:
got rid of that 👍